### PR TITLE
feat(console,phrases): support custom domain in protected app details

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Support custom domain in protected app details page:
1. Refactored tenant custom domain components, in order to make them re-usable in protected app details page.
2. Add custom domain field in protected app details page.

## Testing

image

## Checklist

~~- [ ] `.changeset`~~
~~- [ ] unit tests~~
~~- [ ] integration tests~~
~~- [ ] necessary TSDoc comments~~

### CODE REVIEW

File: packages/toolkit/core-kit/src/regex.ts [hunk 1]
```diff
@@ -7,3 +7,5 @@ export const mobileUriSchemeProtocolRegEx = /^[a-z][\d+_a-z-]*(\.[\d+_a-z-]+)+:$
 export const hexColorRegEx = /^#[\da-f]{3}([\da-f]{3})?$/i;
 export const dateRegex = /^\d{4}(-\d{2}){2}/;
 export const noSpaceRegEx = /^\S+$/;
+export const subdomainRegEx =
```
Comment (wangsijie) [inline packages/toolkit/core-kit/src/regex.ts:10]: There is a similar one in "shared"

**Review by wangsijie — APPROVED** [review]
LGTM

File: packages/toolkit/core-kit/src/regex.ts (hunk 1 above)
Comment (charIeszhao) [reply packages/toolkit/core-kit/src/regex.ts:10]: Let me check..

Comment (charIeszhao) [reply packages/toolkit/core-kit/src/regex.ts:10]: This input field requires full domain URL that has at least 3 parts, e.g. foo.bar.com. But I think I need to change the name of the regex from `subdomainRegEx` to `customDomainRegEx` or `domainRegEx`

**Pull request merged**